src/Bridges/Constraint/bridges/integer_to_zeroone.jl

```diff
@@ -0,0 +1,168 @@
+            nothing,
+        )
+    end
+end
+
+const IntegerToZeroOne{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{IntegerToZeroOneBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::MOI.ModelLike,
+    x::MOI.VariableIndex,
+    ::MOI.Integer,
+) where {T}
+    # !!! info
+    #     Postpone creation until final_touch.
+    return IntegerToZeroOneBridge{T}(x)
+end
+
+function MOI.supports_constraint(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::Type{MOI.VariableIndex},
+    ::Type{MOI.Integer},
+) where {T}
+    return true
+end
+
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:IntegerToZeroOneBridge},
+)
+    return Tuple{Type}[(MOI.ZeroOne,)]
+end
+
+function MOI.Bridges.added_constraint_types(
+    ::Type{IntegerToZeroOneBridge{T}},
+) where {T}
+    return Tuple{Type,Type}[(MOI.ScalarAffineFunction{T}, MOI.EqualTo{T})]
+end
+
+function concrete_bridge_type(
+    ::Type{IntegerToZeroOneBridge{T}},
+    ::Type{MOI.VariableIndex},
+    ::Type{MOI.Integer},
+) where {T}
+    return IntegerToZeroOneBridge{T}
+end
+
+function MOI.get(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintFunction,
+    bridge::IntegerToZeroOneBridge,
+)
+    return bridge.x
+end
+
+function MOI.get(::MOI.ModelLike, ::MOI.ConstraintSet, ::IntegerToZeroOneBridge)
+    return MOI.Integer()
+end
+
+function MOI.delete(model::MOI.ModelLike, bridge::IntegerToZeroOneBridge)
+    MOI.delete(model, bridge.ci)
+    MOI.delete(model, bridge.y)
+    return
+end
+
+function MOI.get(bridge::IntegerToZeroOneBridge, ::MOI.NumberOfVariables)::Int64
+    return length(bridge.y)
+end
+
+function MOI.get(bridge::IntegerToZeroOneBridge, ::MOI.ListOfVariableIndices)
+    return copy(bridge.y)
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge,
+    ::MOI.NumberOfConstraints{MOI.VariableIndex,MOI.ZeroOne},
+)::Int64
+    return length(bridge.y)
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge,
+    ::MOI.ListOfConstraintIndices{MOI.VariableIndex,MOI.ZeroOne},
+)
+    return map(bridge.y) do y
+        return MOI.ConstraintIndex{MOI.VariableIndex,MOI.ZeroOne}(y.value)
+    end
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge{T},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}},
+)::Int64 where {T}
+    return 1
+end
+
+function MOI.get(
+    bridge::IntegerToZeroOneBridge{T},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T},MOI.EqualTo{T}},
+) where {T}
+    return [bridge.ci]
+end
+
+MOI.Bridges.needs_final_touch(::IntegerToZeroOneBridge) = true
+
+function MOI.Bridges.final_touch(
+    bridge::IntegerToZeroOneBridge{T},
+    model::MOI.ModelLike,
+) where {T}
+    ret = MOI.Utilities.get_bounds(model, T, bridge.x)
+    if ret === bridge.last_bounds
+        return nothing  # final_touch already called
+    elseif ret[1] == typemin(T) || ret[2] == typemax(T)
+        error(
+            "Unable to use IntegerToZeroOneBridge because the variable " *
```

Comment:
I am wondering if this should check against `nothing`s? My reasoning is that one could provide these values as the bounds intentionally. I can see that it might have to be done because of the defaults currently but it seems to be an edge case that might come up

Reply:
> My reasoning is that one could provide these values as the bounds intentionally.

Which values ? I'm not sure to follow

Reply:
One of `typemax(T)` or `typemin(T)`

Reply:
Most solvers will treat variables as unbounded if the bounds are > 1e20 or so (sometimes 1e30, sometimes 1e101).

You should never set the bounds as `typemin(T)` or `typemax(T)`. And even if you did, we can't reformulate that as binary.